### ISSUE TITLE
`private_subject` false positives in function scope

### ISSUE THREAD

**Issue by ejensen** [realm/SwiftLint#4643]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Starting in SwiftLint 0.50.0 the `private_subject` rule triggers on function local variables. This was not the case in 0.49.1 and earlier. 

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
  * 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
  * CocoaPods
* Paste your configuration file:

```yml
opt_in_rules:
  - private_subject
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  * No
* Which Xcode version are you using (check `xcodebuild -version`)?
  * 14.1
* Do you have a sample that shows the issue?

```swift
func foo() {
    // This triggers a violation "Private Combine Subject Violation: Combine Subject should be private. (private_subject)":
    let goodSubject = PassthroughSubject<Bool, Never>(true)
}
```
